webpack.config.mjs

```diff
@@ -148,4 +150,21 @@ const workerConfig = {
     }
 };
 
+// Replace dynamic import with a static import, so Webpack can bundle the resources
+for (const workerName of Object.keys(workerConfig.entry)) {
+    const bundlePath = path.join('node_modules', '@yowasp', workerName, 'gen', 'bundle.js');
+    let bundleContent = readFileSync(bundlePath, {
+        encoding: 'utf-8'
+    });
+
+    bundleContent = bundleContent.replaceAll(
+        'import(this.resourceFileURL)',
+        `import(/* webpackMode: "eager" */ './resources-${workerName}.js')`
```

Comment:
Would it help if I changed the upstream code where now it does:
```
var resourceFileURL = new URL("./resources-nextpnr-ice40.js", import.meta.url);
var icepll = new Application(resourceFileURL, instantiate, "yowasp-icepll");
```
to be instead:
```
var resources = lazy(() => import("./resources-nextpnr-ice40.js"));
var icepll = new Application(resources, instantiate, "yowasp-icepll");
```
or something like that?

Reply:
If it helps webpack resolve the module then I do believe that would be better. This solution was supposed to fix the webpack config mess haha

Reply:
@whitequark I will test the proposed code locally this week. I'm pretty confident it will solve the issue I was having with Webpack bundling.

Reply:
@whitequark I tested it and helps a bit. I found that I still have to insert the `/* webpackMode: "eager" */` comment, which is obliviously not desirable to include in the libraries. What was the reason for using a dynamic import instead of using an import statement similar to `import { instantiate} from '../gen/yosys.js'` ?

Reply:
> What was the reason for using a dynamic import instead of using an import statement similar to `import { instantiate} from '../gen/yosys.js'` ?

I would like to absolutely minimize the amount of code loaded by dynamically importing `yosys.js` because I have a use case in [my extension](https://marketplace.visualstudio.com/items?itemName=yowasp.toolchain) where I load and retrieve the list of commands from a large set of bundles (roughly 200 MB of code and data, the entire FOSS toolchain). I am trying to minimize the bandwidth used by this operation since it would be impractical to get user consent to download this much data in my application. I hope this makes sense.

> I found that I still have to insert the `/* webpackMode: "eager" */` comment, which is obliviously not desirable to include in the libraries.

I think I can add that to the libraries. In the case where someone is using webpack they have pre-determined through their design process that using such amount of bandwidth is OK and therefore don't need the lazy loading. Is that OK with you?

Reply:
I would not include the Webpack comment in the YoWASP libraries. It's pretty specific to our setup. I'm fine with having the string replace code in our build script.

Reply:
No problem. Should I change the import statement?

Reply:
Yes, that would be helpful.

Reply:
Done. Yosys is already [released](https://www.npmjs.com/package/@yowasp/yosys/v/0.38.21-dev.654) and you should see a nextpnr release soon. Also, eventually [WASI-Virt](https://github.com/bytecodealliance/WASI-Virt) will replace `resources.js` entirely (right now it's a bit [broken](https://github.com/bytecodealliance/WASI-Virt/issues/37)).

Reply:
Thanks.